web/packages/design/src/Label/LabelButtonWithIcon.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import styled from 'styled-components';
+
+import Flex from 'design/Flex';
+import { IconProps } from 'design/Icon/Icon';
+
+import Label, { LabelProps } from './Label';
+import { IconPlacement } from './types';
+
+export function LabelButtonWithIcon({
+  Icon,
+  placement,
+  children,
+  ...labelProps
+}: LabelProps & {
+  Icon: React.ComponentType<IconProps>;
+  placement: IconPlacement;
+  children: React.ReactNode;
+  onClick?: () => void;
+}) {
+  const icon = (
+    <ButtonIcon>
+      <Icon size="small" />
+    </ButtonIcon>
+  );
+
+  return (
+    <LabelWithHoverAffect {...labelProps} withHoverState>
```

Comment:
Support keyboard navigation.

```suggestion
    <LabelWithHoverAffect {...labelProps} withHoverState tabIndex={0}>
```

Also, consider adding `:focus` and/or `:active` styles.

Reply:
> Also, consider adding :focus and/or :active styles.

I'll punt this with a follow up, i have to ask samrat how to color those states